modules/transport/Transport.js

```diff
@@ -0,0 +1,239 @@
+import {
+    MESSAGE_TYPE_EVENT,
+    MESSAGE_TYPE_RESPONSE,
+    MESSAGE_TYPE_REQUEST
+} from './constants';
+
+/**
+ * Stores the currnet transport that have to be used.
+ */
+export default class Transport {
+    /**
+     * Creates new instance.
+     *
+     * @param {Object} options - Optional parameters for configuration of the
+     * transport.
+     */
+    constructor({ transport } = {}) {
```

Comment:
Continuing the renaming in the name of readability that I started in another comment:
- `transport` -> `backend`.

Generally, the constructor of `Transport` receiving `transport` that's actually an instance of something that ends with `TransportBackend` is unclear enough to me.